deeppavlov/core/layers/embeddings.py

```diff
@@ -0,0 +1,141 @@
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+
+        self.vocab_size = vocab_size
+        self.token_type_vocab_size = token_type_vocab_size
+        self.sep_token_index = sep_token_index
+        self.output_dim = output_dim
+        self.max_len = max_len
+        self.embeddings_initializer = tf.keras.initializers.TruncatedNormal(stddev=initializer_range)
+        self.trainable_pos_embedding = trainable_pos_embedding
+
+        self.supports_masking = True
+
+    @tf_utils.shape_type_conversion
+    def build(self, batch_input_shape: Union[Sequence[int], tf.TensorShape]) -> None:
+        # Note: most sparse optimizers do not have GPU kernels defined. When
+        # building graphs, the placement algorithm is able to place variables on CPU
+        # since it knows all kernels using the variable only exist on CPU.
+        # When eager execution is enabled, the placement decision has to be made
+        # right now. Checking for the presence of GPUs to avoid complicating the
+        # TPU codepaths which can handle sparse optimizers.
+        if context.executing_eagerly() and context.context().num_gpus():
+            with ops.device('cpu:0'):
+                self._create_weights(batch_input_shape)
+        else:
+            self._create_weights(batch_input_shape)
+        self.built = True
+
+    def _create_weights(self, batch_input_shape: Union[Sequence[int], tf.TensorShape]) -> None:
+
+        self.token_emb_table = self.add_weight(shape=(self.vocab_size, self.output_dim),
+                                               dtype=tf.float32,
+                                               initializer=self.embeddings_initializer,
+                                               name='word_embeddings')
+
+        self.token_type_emb_table = self.add_weight(shape=(self.token_type_vocab_size, self.output_dim),
+                                                    dtype=tf.float32,
+                                                    initializer=self.embeddings_initializer,
+                                                    name='token_type_embeddings')
+
+        # Since the position embedding table is a learned variable, we create it using a (long) sequence length
+        # `max_len`. The actual sequence length might be shorter than this, for faster training of
+        # tasks that do not have long sequences.
+        self.full_position_emb_table = self.add_weight(shape=(self.max_len, self.output_dim),
+                                                       dtype=tf.float32,
+                                                       initializer=self.embeddings_initializer,
+                                                       trainable=self.trainable_pos_embedding,
+                                                       name='position_embeddings')
+
+    def call(self,
+             token_ids: tf.Tensor,
+             **kwargs) -> tf.Tensor:
```

Comment:
It would be better if this method had a docstring that would at least tell us the shapes of output.

Reply:
Ok, I will add. However, there is standard comput_output_shape method for shapes :)